### ISSUE TITLE
`addChildNodes` fails after `removeNode` is called on Tree with single Node

### ISSUE THREAD

**Issue by mudcube** [cheton/infinite-tree#32]
Reproduce with the following code:

```js
var i = new InfiniteTree()
i.addChildNodes({id: 'test'})
console.log(i.nodes)
i.removeNode(i.nodes[0])
console.log(i.nodes)
i.addChildNodes({id: 'test'})
console.log(i.nodes)
```

Results in:

```
[Node]
[]
[]
```

Should result in:

```
[Node]
[]
[Node]
```

**Comment by cheton** [comment]
The key problem is that the value of `rootNode.state.open` should always be true even when removing the child:

![image](https://user-images.githubusercontent.com/447801/40853396-888fa4ac-6600-11e8-8fdb-78fe41169732.png)

Your fix looks good to me, but I'd like to make below change by comparing `parentNode` with the `rootNode` rather than checking empty parent to improve code readability:

```js
 if (parentNode !== this.state.rootNode) {
    parentNode.state.open = parentNode.state.open && (parentNode.children.length > 0);
 }
```

**Comment by cheton** [comment]
Fixed in [v1.16.1](https://github.com/cheton/infinite-tree/releases/tag/v1.16.1)
